server/schedulers/hot_region.go

```diff
@@ -444,16 +429,8 @@ func (bs *balanceSolver) isValid() bool {
 	if bs.Cluster == nil || bs.sche == nil || bs.stLoadDetail == nil {
 		return false
 	}
-	switch bs.rwTy {
-	case statistics.Write, statistics.Read:
-	default:
-		return false
-	}
-	switch bs.opTy {
-	case movePeer, transferLeader:
-	default:
-		return false
-	}
+	// ignore the return value because it will panic if the type is not correct.
+	_ = toResourceType(bs.rwTy, bs.opTy)
```

Comment:
It returns false if the type is wrong in origin code, and I think `toResourceType` should not  panic.

Reply:
In the original code, it will panic inside the `init` function.